test/e2e/helpers_test.go

```diff
@@ -528,31 +527,48 @@ func verifyIngressWithEchoBackendsInAdminAPI(
 	t.Helper()
 
 	require.Eventually(t, func() bool {
-		d, err := dump.Get(ctx, kongClient, dump.Config{})
+		start := time.Now()
+		defer func() {
+			t.Logf("Try fetching config from %q started at %s, duration %v", kongClient.BaseRootURL(), start.Format(time.RFC3339), time.Since(start))
+		}()
+
+		services, err := kongClient.Services.ListAll(ctx)
 		if err != nil {
-			t.Logf("failed dumping config: %s", err)
+			t.Logf("failed to list services: %v", err)
 			return false
 		}
-		if len(d.Services) != 1 {
+		if len(services) != 1 || services[0].ID == nil {
 			t.Log("still no service found...")
 			return false
 		}
-		if len(d.Routes) != 1 {
+
+		routes, _, err := kongClient.Routes.ListForService(ctx, services[0].ID, &kong.ListOpt{Size: 10})
+		if err != nil {
+			t.Logf("failed to list routes for service %s: %v", *services[0].ID, err)
+			return false
+		}
+		if len(routes) != 1 {
 			t.Log("still no route found...")
 			return false
 		}
-		if d.Services[0].ID == nil ||
-			d.Routes[0].Service.ID == nil ||
-			*d.Services[0].ID != *d.Routes[0].Service.ID {
-			t.Log("still no matching service found...")
+
+		upstreams, err := kongClient.Upstreams.ListAll(ctx)
+		if err != nil {
+			t.Logf("failed to list upstreams: %v", err)
+			return false
+		}
+		if len(upstreams) != 1 || upstreams[0].ID == nil {
+			t.Logf("still no upstreams found...")
 			return false
 		}
-		if len(d.Targets) != noReplicas {
-			t.Log("still no target found...")
+
+		targets, err := kongClient.Targets.ListAll(ctx, upstreams[0].ID)
+		if err != nil {
+			t.Logf("failed to list targets for upstream %s: %v", *upstreams[0].ID, err)
 			return false
 		}
-		if len(d.Upstreams) != 1 {
-			t.Log("still no upstream found...")
+		if len(targets) != noReplicas {
+			t.Log("still no targets found...")
```

Comment:
We might want to add a bit more debugging info. Something along the lines:

```suggestion
			t.Logf("expected %d targets, found %d", noReplicas, len(targets))
```